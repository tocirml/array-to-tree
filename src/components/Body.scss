.App-body {
  padding: 5px;
  .body-message {
    margin: 10px;
  }
  .OutputContainer {
    padding: 10px;
    border: 1px solid black;
    background: rgb(70, 69, 69);
    overflow-x: scroll;
    left: 3px;
  }
}
