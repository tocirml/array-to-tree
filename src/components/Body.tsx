import * as React from 'react';
import { IAppState } from '../store/IAppState';
import { observer } from 'mobx-react';
import { TreeInput } from './TreeInput';
import { TreeOutput } from './TreeOutput';
import './Body.scss';
import { useAppStateContext } from '../store/AppState';

interface BodyProps {
  appState: IAppState;
}

const BodyRenderer: React.FunctionComponent<BodyProps> = observer(props => {
  const changeHandler = (newVal: any): void => {
    props.appState.setState({
      ...props.appState,
      treeNode: newVal,
    });
  };

  return (
    <main className="App-body">
      {props.appState!.bodyMessage}
      <TreeInput onChange={changeHandler} />
      <div className="OutputContainer">
        <TreeOutput treeNode={props.appState.treeNode} />
      </div>
    </main>
  );
});

export const Body: React.FunctionComponent<{}> = props => {
  const appState = useAppStateContext();
  return <BodyRenderer appState={appState} />;
};

export default Body;
