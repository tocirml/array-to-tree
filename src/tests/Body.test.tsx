import React from 'react';
import Body, { BodyRenderer } from '../components/Body';
import { shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import '@testing-library/jest-dom/extend-expect';

const BODY_MESSAGE = 'Process the input text to a BinTreeNode';

describe('rendering', () => {
  it('renders without crashing', () => {
    render(<Body />);
  });

  it('renders correct body message', () => {
    render(<Body />);
    expect(screen.getByTestId('body-message')).toHaveTextContent(BODY_MESSAGE);
  });
});

// describe('functionality', () => {
//     it('calls changeHandler on fetch', () => {
//         const changeHandler = jest.fn();

//     })
// })

describe('snapshots', () => {
  it('Body snapshot', () => {
    const tree = shallow(<Body />);
    expect(tree).toMatchSnapshot();
  });

  it('BodyRenderer snapshot', () => {
    const appState: any = {
      appState: {
        bodyMessage: BODY_MESSAGE,
      },
    };
    const tree = shallow(<BodyRenderer appState={appState} />);
    expect(tree).toMatchSnapshot();
  });
});
