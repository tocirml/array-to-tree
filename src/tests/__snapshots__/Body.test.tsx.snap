// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Body snapshot 1`] = `
<Memo(wrappedComponent)
  appState={
    AppState {
      "bodyMessage": "Process the input text to a BinTreeNode",
      "smallestTreeId": "root",
      "title": "Tree Traversal",
      "treeNode": BinTreeNode {
        "id": "root",
        "left": null,
        "right": null,
      },
    }
  }
/>
`;

exports[`snapshots BodyRenderer snapshot 1`] = `
<main
  className="App-body"
>
  <div
    className="body-message"
    data-testid="body-message"
  />
  <TreeInput
    onChange={[Function]}
  />
  <TreeTextArea
    onChange={[Function]}
  />
  Output:
  <div
    className="OutputContainer"
  >
    <TreeOutput />
  </div>
</main>
`;
